<launch>
  <arg name="play_bag" default="true" />
  <arg name="bag_file" default="/home/zhiyuan/Documents/IP_Data/Raw/metal_stairs_2019-01-17-16-53-00.bag"/> 
  <arg name="voxel_size" default="0.05"/>

  <!-- Download the dataset here: https://projects.asl.ethz.ch/datasets/doku.php?id=iros2017 -->

  <node name="player" pkg="rosbag" type="play" output="screen" args="-r 1.0 --clock --pause $(arg bag_file)" if="$(arg play_bag)"/>

   <param name="/use_sim_time" value="true" />
   <arg name="manager" default="test_nodelet_manager"/>
   <group ns="gripper_rgbd_cam">
     <node pkg="nodelet" type="nodelet" name="$(arg manager)" args="manager"
           output="screen">
     </node>
     <include file="$(find rgbd_launch)/launch/includes/processing.launch.xml">
       <arg name="manager" default="$(arg manager)"/>
       <arg name="rgb" value="color" />
      </include>
  </group>

  <node name="relay_depth_image"  pkg="topic_tools" type="relay"  
    args="/gripper_rgbd_cam/depth/image_rect_raw/gripper_rgbd_cam/depth/image_rect" respawn="true" />

   
	
   <node name="voxblox_node" pkg="voxblox_ros" type="esdf_server" output="screen" args="-alsologtostderr" clear_params="true">
    <remap from="pointcloud" to="/gripper_rgbd_cam/depth_registered/points"/> 
    <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
    <param name="tsdf_voxels_per_side" value="16" />
    <param name="voxel_carving_enabled" value="true" />
    <param name="color_mode" value="color" />
    <param name="use_tf_transforms" value="false" />
    <param name="update_mesh_every_n_sec" value="1.0" />
    <param name="min_time_between_msgs_sec" value="0.0" />
    <param name="method" value="fast" />
    <param name="use_const_weight" value="false" />
    <param name="allow_clear" value="true" />
    <param name="verbose" value="true" />
    <param name="world_frame" value="odom"/> 
    <rosparam file="$(find voxblox_ros)/cfg/cow_and_lady.yaml"/>
    <param name="mesh_filename" value="$(find voxblox_ros)/mesh_results/$(anon cow).ply" />
  </node>

</launch>
